src/presence/presenceModel.js

```diff
@@ -161,3 +198,70 @@ export const getPresenceStatusForUserId = (
     ? statusFromPresence(userPresence)
     : statusFromPresenceAndUserStatus(userPresence, userStatus);
 };
+
+//
+//
+// Reducer.
+//
+
+const initialState: PresenceState = {
+  byEmail: Immutable.Map(),
```

Comment:
nit, from style guide:

> **Always provide a type when writing an empty `Immutable` value**:
> Whenever you create an empty `Immutable.Map`, `Immutable.List`, or
> so on, specify the intended type explicitly.  For example:
> ```js
> const map: Immutable.Map<number, string> = Immutable.Map(); // good
> 
> const map = Immutable.Map(); // BAD -- don't do
> ```
> 
> This is essential in order to get effective type-checking of the
> code that uses the new collection.  (It's not clear if this is a bug
> in Flow, or a design limitation of Flow, or an issue in the Flow types
> provided by Immutable.js, or some combination.)

I think it could look like this:

```diff
diff --git src/presence/presenceModel.js src/presence/presenceModel.js
index a46559acb..a9f112efa 100644
--- src/presence/presenceModel.js
+++ src/presence/presenceModel.js
@@ -205,7 +205,7 @@ export const getPresenceStatusForUserId = (
 //
 
 const initialState: PresenceState = {
-  byEmail: Immutable.Map(),
+  byEmail: Immutable.Map<string, UserPresence>(),
 };
```

Reply:
Thanks, I'd forgotten to consider that.

I think this is covered, though, by the type annotation on the object as a whole — that `: PresenceState`. That's explicit in much the same way as the annotation in the "good" example.

And you can confirm it's effective by writing an expression like `initialState.byEmail.values()` and hovering in the IDE, to see it returns an iterator of `UserPresence`.